meteor/client/ui/Shelf/MiniRundownPanel.tsx

```diff
@@ -0,0 +1,237 @@
+import * as React from 'react'
+import * as _ from 'underscore'
+import ClassNames from 'classnames'
+import {
+	RundownLayoutBase,
+	DashboardLayoutMiniRundown,
+	RundownLayoutMiniRundown,
+	DashboardLayoutNextInfo,
+} from '../../../lib/collections/RundownLayouts'
+import { RundownLayoutsAPI } from '../../../lib/api/rundownLayouts'
+import { withTracker } from '../../lib/ReactMeteorData/ReactMeteorData'
+import { MeteorReactComponent } from '../../lib/MeteorReactComponent'
+import { RundownPlaylist } from '../../../lib/collections/RundownPlaylists'
+import { PartInstance, PartInstanceId, PartInstances } from '../../../lib/collections/PartInstances'
+import { Segment, Segments } from '../../../lib/collections/Segments'
+import { dashboardElementPosition } from './DashboardPanel'
+import { Part } from '../../../lib/collections/Parts'
+import { unprotectString } from '../../../lib/lib'
+import { Meteor } from 'meteor/meteor'
+
+interface IMiniRundownPanelProps {
+	key: string
+	visible?: boolean
+	layout: RundownLayoutBase
+	panel: RundownLayoutMiniRundown
+	playlist: RundownPlaylist
+}
+
+interface IMiniRundownPanelTrackedProps {
+	currentSegment?: Segment
+	currentPartInstance?: PartInstance
+	nextPartInstance?: PartInstance
+	nextSegment?: Segment
+	allParts?: Part[]
+	allSegments?: Segment[]
+}
+
+interface IState {}
+
+interface MiniRundownPart {
+	identifier: any
+	segmentName: string
+	partName: string
+	cssClass: string
+}
+
+export class MiniRundownPanelInner extends MeteorReactComponent<
+	IMiniRundownPanelProps & IMiniRundownPanelTrackedProps,
+	IState
+> {
+	constructor(props) {
+		super(props)
+		this.state = {}
+	}
+
+	componentDidUpdate() {
+		Meteor.setTimeout(() => {
+			const el = document.getElementById(getCurrentPartIdName())
+			if (el) {
+				el.scrollIntoView({
+					behavior: 'smooth',
+				})
+			}
+		}, 1000)
+	}
+
+	render() {
+		const isDashboardLayout = RundownLayoutsAPI.isDashboardLayout(this.props.layout)
+		const style = getElementStyle(this.props, isDashboardLayout)
+
+		const miniRundowns: MiniRundownPart[] = getMiniRundownList(
+			this.props.allParts,
+			this.props.allSegments,
+			this.props.currentPartInstance,
+			this.props.nextPartInstance
+		)
+
+		return (
+			<div
+				key={'miniRundownContainer'}
+				className={ClassNames('dashboard-panel mini-rundown-panel', getContainerClass(this.props, isDashboardLayout))}
+				style={getContainerStyle(this.props, isDashboardLayout)}
+			>
+				<span className="mini-rundown-panel__name" style={style} key={'miniRundownHeader'}>
+					{this.props.panel.name}{' '}
+				</span>
+
+				{miniRundowns.map((miniRundown: MiniRundownPart, index: number) => (
```

Comment:
If we are having a label for the component "Mini rundown", then I would like to wrap the actual rundown in a div, such that the label is excluded from the scrolling area and is visible at all times.

Reply:
![image](https://user-images.githubusercontent.com/91883429/154698130-3c4b60cb-16af-4e64-be01-8a1c8c165879.png)